internal/gbimporter/gbimporter.go

```diff
@@ -81,7 +91,21 @@ func (i *importer) ImportFrom(path, srcDir string, mode types.ImportMode) (*type
 	def.SplitPathList = i.splitPathList
 	def.JoinPath = i.joinPath
 
-	return i.underlying.ImportFrom(path, srcDir, mode)
+	pkg, err := i.underlying.ImportFrom(path, srcDir, mode)
+	if err != nil {
+		return nil, err
+	}
+
+	// add the package to the cache
+	i.cache[path] = pkg
+	// delete the pakcage after ttl expires
+	timer := time.NewTimer(time.Duration(i.ttl) * time.Minute)
```

Comment:
@anjmao does this cache cause any problems in https://github.com/sourcegraph/go-langserver/pull/305?

Maybe it should be possible to disable caching if `ttl == 0`?

Reply:
@lloiser  Langserver has it's own global cache and resets it on file changes https://github.com/sourcegraph/go-langserver/blob/master/langserver/handler.go#L429. 

In my PR I didn't used any cache from langserver yet except go package binary cache. I'm thinking to add it in separate PR and make gocode internal package stateless which does not do any caching inside but instead uses langserver cache.